pkg/api/v20230701preview/openshiftcluster_validatestatic.go

```diff
@@ -228,6 +231,44 @@ func (sv openShiftClusterStaticValidator) validateNetworkProfile(path string, np
 			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".outboundType", "The provided outboundType '%s' is invalid: cannot use UserDefinedRouting if either API Server Visibility or Ingress Visibility is public.", np.OutboundType)
 		}
 	}
+
+	if np.OutboundType == OutboundTypeUserDefinedRouting && np.LoadbalancerProfile != nil {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".loadBalancerProfile", "The provided loadBalancerProfile is invalid: cannot use a loadBalancerProfile if outboundType is UserDefinedRouting.")
+	}
```

Comment:
So today, we have UDR and Loadbalancer, right? Would it be better for us to strictly check for LB type network instead of "not UDR" here in the case where we add additional types later? That seems a little less brittle to me.

Reply:
So you are thinking something like this?  `np.OutboundType == OutboundTypeLoadbalancer && np.LoadbalancerProfile == nil` then return error?  The issue with this approach is the LoadbalancerProfile is not mandatory during cluster creation, so the cluster doc may not have a LoadbalancerProfile until `api.SetDefaults` is called, which does not happen until AFTER static validation.

Reply:
I think I would rewrite it as `if np.LoadbalancerProfile != nil && np.OutboundType != OutboundTypeLoadbalancer` or whatever it's called.

Reply:
It would have to be `np.LoadbalancerProfile != nil && np.OutboundType != OutboundTypeLoadbalancer && np.OutboundType != ""` because OutboundType is not mandated to be set and if it is an empty string it will default to `OutboundTypeLoadbalancer`.  If defaults were ever changed then this would break, but I don't know if/when we would ever change the default value as that would be a breaking change within itself...

Reply:
We talked about this a bit on slack, so bringing the jist of our conversation here - the way that our API is architected, we set defaults after static validation, and if we did the `!= ""` check in static validation here, that would be the first of its kind.

On my side, I think I'm ok to say "if you want an LB Profile, you've gotta strictly supply networkType=LB, instead of relying on our (current) default", though, in AKS, this will work for you (empty networktype, populated LB Profile), I assume because they set defaults before doing validation...